From: martin f. krafft <madduck@debian.org>
Subject: Set /etc/mdadm/mdadm.conf as primary config file location

On Debian, the configuration file resides primarily in /etc/mdadm/mdadm.conf,
/etc/mdadm.conf is only used as a backup.

This is a Debian-specific patch.

Forwarded: not-needed
Reviewed-by: martin f. krafft <madduck@debian.org>

---
 Makefile     |    4 ++--
 ReadMe.c     |    2 +-
 mdadm.8      |   14 ++++++--------
 mdadm.conf.5 |    2 +-
 mdassemble.8 |    2 +-
 5 files changed, 11 insertions(+), 13 deletions(-)

diff --git a/Makefile b/Makefile
index 5636392..2b8b17c 100644
--- a/Makefile
+++ b/Makefile
@@ -54,8 +54,8 @@ CPPFLAGS=
 endif
 
 SYSCONFDIR = /etc
-CONFFILE = $(SYSCONFDIR)/mdadm.conf
-CONFFILE2 = $(SYSCONFDIR)/mdadm/mdadm.conf
+CONFFILE = $(SYSCONFDIR)/mdadm/mdadm.conf
+CONFFILE2 = $(SYSCONFDIR)/mdadm.conf
 MAILCMD =/usr/sbin/sendmail -t
 CONFFILEFLAGS = -DCONFFILE=\"$(CONFFILE)\" -DCONFFILE2=\"$(CONFFILE2)\"
 CFLAGS = $(CWFLAGS) $(CXFLAGS) -DSendmail=\""$(MAILCMD)"\" $(CONFFILEFLAGS)
diff --git a/ReadMe.c b/ReadMe.c
index 1c118f2..ac1f3c0 100644
--- a/ReadMe.c
+++ b/ReadMe.c
@@ -549,7 +549,7 @@ char Help_incr[] =
 ;
 
 char Help_config[] =
-"The /etc/mdadm.conf config file:\n\n"
+"The /etc/mdadm/mdadm.conf config file:\n\n"
 " The config file contains, apart from blank lines and comment lines that\n"
 " start with a hash(#), four sorts of configuration lines: array lines, \n"
 " device lines, mailaddr lines and program lines.\n"
diff --git a/mdadm.8 b/mdadm.8
index 676a9cd..4bef108 100644
--- a/mdadm.8
+++ b/mdadm.8
@@ -267,9 +267,9 @@ the exact meaning of this option in different contexts.
 .TP
 .BR \-c ", " \-\-config=
 Specify the config file.  Default is to use
-.BR /etc/mdadm.conf ,
-or if that is missing then
-.BR /etc/mdadm/mdadm.conf .
+.BR /etc/mdadm/mdadm.conf ,
+or if that is missing, then
+.BR /etc/mdadm.conf .
 If the config file given is
 .B "partitions"
 then nothing will be read, but
@@ -1326,8 +1326,6 @@ The config file is only used if explicitly named with
 or requested with (a possibly implicit)
 .BR \-\-scan .
 In the later case,
-.B /etc/mdadm.conf
-or
 .B /etc/mdadm/mdadm.conf
 is used.
 
@@ -1649,7 +1647,7 @@ or
 .B \-\-scan
 will cause the output to be less detailed and the format to be
 suitable for inclusion in
-.BR /etc/mdadm.conf .
+.BR /etc/mdadm/mdadm.conf .
 The exit status of
 .I mdadm
 will normally be 0 unless
@@ -1704,7 +1702,7 @@ or
 is given, then multiple devices that are components of the one array
 are grouped together and reported in a single entry suitable
 for inclusion in
-.BR /etc/mdadm.conf .
+.BR /etc/mdadm/mdadm.conf .
 
 Having
 .B \-\-scan
@@ -2330,7 +2328,7 @@ uses this to find arrays when
 is given in Misc mode, and to monitor array reconstruction
 on Monitor mode.
 
-.SS /etc/mdadm.conf
+.SS /etc/mdadm/mdadm.conf
 
 The config file lists which devices may be scanned to see if
 they contain MD super block, and gives identifying information
diff --git a/mdadm.conf.5 b/mdadm.conf.5
index 002e2b3..efe066f 100644
--- a/mdadm.conf.5
+++ b/mdadm.conf.5
@@ -8,7 +8,7 @@
 .SH NAME
 mdadm.conf \- configuration for management of Software RAID with mdadm
 .SH SYNOPSIS
-/etc/mdadm.conf
+/etc/mdadm/mdadm.conf
 .SH DESCRIPTION
 .PP
 .I mdadm
diff --git a/mdassemble.8 b/mdassemble.8
index 5520e8e..37ddca4 100644
--- a/mdassemble.8
+++ b/mdassemble.8
@@ -40,7 +40,7 @@ There are no options to
 
 .SH FILES
 
-.SS /etc/mdadm.conf
+.SS /etc/mdadm/mdadm.conf
 
 The config file lists which devices may be scanned to see if
 they contain MD super block, and gives identifying information
-- 
tg: (d28c1a7..) debian/conffile-location (depends on: upstream)
